events/views.py

```diff
@@ -2185,20 +2187,22 @@ def test_attendance(request, tid):
             """)
     mdlids = []
     participant_ids = []
-    online_participant_ids = list(TestAttendance.objects.filter(test_id = test.id).values_list('mdluser_id'))
-    for k in online_participant_ids:
-        mdlids.append(k[0])
-
-    if test.test_category_id == 1:
-        participant_ids = list(TrainingAttendance.objects.filter(training_id = test.training_id).values_list('mdluser_id'))
-    elif test.test_category_id == 2:
-        participant_ids = list(TrainingAttendance.objects.filter(training_id = test.training_id).values_list('mdluser_id'))
+    if ta_status:
+        mdlids = list(TestAttendance.objects.filter(test_id = test.id, status=ta_status).values_list('mdluser_id', flat=True))
+    else:
+        mdlids = list(TestAttendance.objects.filter(test_id = test.id).values_list('mdluser_id', flat=True))
+    WORKSHOP = 1
+    TRAINING = 2
+    if test.test_category_id in [WORKSHOP, TRAINING]:
+        participant_ids = list(TrainingAttendance.objects.filter(training_id = test.training_id).values_list('mdluser_id', flat=True))
     else:
-        participant_ids = list(TestAttendance.objects.filter(test_id = test.id).values_list('mdluser_id'))
+        if ta_status:
+            participant_ids = list(TestAttendance.objects.filter(test_id = test.id, status=ta_status).values_list('mdluser_id', flat=True))
+        else:
+            participant_ids = list(TestAttendance.objects.filter(test_id = test.id).values_list('mdluser_id', flat=True))
 
     wp = None
-    for k in participant_ids:
-        mdlids.append(k[0])
+    mdlids = mdlids + participant_ids
```

Comment:
```suggestion
    mdlids.extend(participant_ids)
```
use extend to append one array to another

Reply:
Implemented the suggestion